### PR TITLE
Document PDF files extension concepts

### PR DESCRIPTION
**Pull request by mickaelandrieu**
- [x] Templates management
- [x] Variables management
- [x] Available hooks

### CODE REVIEW

File: src/content/1.7/modules/concepts/pdf.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+{assign var=font_size_text value="12pt"}
+{assign var=font_size_header value="12pt"}
+{assign var=font_size_product value="12pt"}
+{assign var=height_header value="25px"}
+{assign var=table_padding value="7px"}
+```
+
+You will get an Invoice would look like this:
+
+{{< figure src="../img/invoice_style.png" title="Stylized Invoice exemple" >}}
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/pdf.md:138]:
```suggestion
{{< figure src="../img/invoice-style.png" title="Stylized Invoice exemple" >}}
```
Avoid underscore for SEO purpose

Comment (mickaelandrieu) [reply src/content/1.7/modules/concepts/pdf.md:138]: Nani ? :neutral_face:

Comment (matks) [reply src/content/1.7/modules/concepts/pdf.md:138]: Yes. Apparently (I did not know this morning) SEO is better for dashes than for underscores. And we want the devdocs, images included, to be SEO boosted in order to make it easier for developers to find their answers when googling an issue.

You can check it 😉

Comment (mickaelandrieu) [reply src/content/1.7/modules/concepts/pdf.md:138]: I don't disagree on this, but it's not a schema it's a screenshot.

Anyway, I'll update.

Can I have a review on the content now, please ? 🙏

File: src/content/1.7/modules/concepts/pdf.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+You can alter the rendering using the provided data or push the PDF Rendering to its best, introducing your very own information that will made your PDF files amazing.
+
+## Customize the PDF files templates
+
+All PDF generated by PrestaShop are using Smarty templates and TCPDF as PDF rendering engine.
+
+You will find all theses templates in the `pdf` folder, for each use case
+managed natively by the Core.
+
+You can alter the Core files OR override them by copying them on `themes/<your-themes>/pdf` folder.
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:23]:
```suggestion
You can alter the Core files OR override them by copying them on `themes/<your-theme>/pdf` folder.
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+pdf
+├── order-slip.payment-tab.tpl
+├── order-slip.product-tab.tpl
+├── order-slip.summary-tab.tpl
+├── order-slip.total-tab.tpl
+└── order-slip.tpl
+```
+
+### Supply Orders
+
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:95]:
```suggestion
{{< minver v="1.7.0" title="true" >}}

```

File: src/content/1.7/modules/concepts/pdf.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+These files are used by most of the previous files:
+
+```
+pdf
+├── footer.tpl
+├── header.tpl
+└── pagination.tpl
+```
+
+### Minimal Exemple: customize design of the Invoices
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:120]:
```suggestion
### Minimal Example: customize design of the Invoices
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+{assign var=font_size_text value="12pt"}
+{assign var=font_size_header value="12pt"}
+{assign var=font_size_product value="12pt"}
+{assign var=height_header value="25px"}
+{assign var=table_padding value="7px"}
+```
+
+You will get an Invoice would look like this:
+
+{{< figure src="../img/invoice-style.png" title="Stylized Invoice exemple" >}}
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:138]:
```suggestion
{{< figure src="../img/invoice-style.png" title="Stylized Invoice example" >}}
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+## Alter or Add information available into PDF templates
+
+Remember the main PDF files we have listed in the previous sections ?
+
+A dynamic hook allows you to alter or add more information that will become available in the related templates.
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:144]:
```suggestion
A dynamic hook allows you to alter or add more information that will become available in the previous templates.
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+
+There is the list of available Template Classes in the Core:
+
+* `HTMLTemplateDeliverySlip`
+* `HTMLTemplateInvoice`
+* `HTMLTemplateOrderReturn`
+* `HTMLTemplateOrderSlip`
+* `HTMLTemplateSupplyOrderForm`
+
+> If your shop is available for the European Union, the GDPR module *must* be installed
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:154]:
```suggestion
> If your shop is available for the European Union, the GDPR module *must* be installed.
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+> this means you can generate another PDF thanks to the class named `HTMLTemplatePSGDPRModule`. This PDF will contains all the Customer Information.
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:155]:
```suggestion
> This means you can generate another PDF thanks to the class named `HTMLTemplatePSGDPRModule`. This PDF will contains all the Customer Information.
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+| Concept       | Hook name                 |
+|---------------|---------------------------|
+| Invoice       | displayPDFInvoice         |
+| Delivery Slip | displayPDFDeliverySlip    |
+| Order Return  | displayPDFOrderReturn     |
+| Order Slip    | displayPDFOrderSlip       |
+| Supply Order  | displayPDFSupplyOrderForm |
+| RGPD Archive  | displayPDFPSGDPRModule    |
+
+### Minimal Exemple: add an extra property to the Invoice
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:176]:
```suggestion
### Minimal Example: add an extra property to the Invoice
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+
+In `modules/your-module/your-module.php`:
+
+```php
+
+
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:182]:
```suggestion
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+public function hookDisplayPDFInvoice($hookArgs)
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:183]:
```suggestion
```php
public function hookDisplayPDFInvoice($hookArgs)
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+
+They are set in the ``PaymentModule:validateOrder`` function.
+
+{{% notice tip %}}
+This is also the right place to `dump` them and figure out what they provide.
+{{% /notice %}}
+
+In this section, will we learn how to alter or add our own Smarty variables.
+
+### Add or alter Smarty variables
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:270]:
```suggestion
### Add or alter Smarty variables
```

**Comment by mickaelandrieu** [comment]
Hello @Progi1984,

thanks for your review, I've applied all your suggestions.

Cheers

**Comment by matks** [comment]
Thank you @mickaelandrieu

**Pull request merged**